satellite/analytics/hubspot.go

```diff
@@ -0,0 +1,197 @@
+		lastName = names[1]
+	} else {
+		firstName = fullName
+	}
+
+	createUser := HubSpotEvent{
+		Endpoint: "https://api.hubapi.com/crm/v3/objects/contacts?hapikey=" + q.escapedAPIKey,
+		Data: map[string]interface{}{
+			"email": fields.Email,
+			"properties": map[string]interface{}{
+				"email":          fields.Email,
+				"firstname":      firstName,
+				"lastname":       lastName,
+				"lifecyclestage": "customer",
+			},
+		},
+	}
+
+	sendUserEvent := HubSpotEvent{
+		Endpoint: "https://api.hubapi.com/events/v3/send?hapikey=" + q.escapedAPIKey,
+		Data: map[string]interface{}{
+			"email":     fields.Email,
+			"eventName": eventPrefix + "_" + "account_created_new",
+			"properties": map[string]interface{}{
+				"userid":             fields.ID.String(),
+				"email":              fields.Email,
+				"name":               fields.FullName,
+				"satellite_selected": q.satelliteName,
+				"account_type":       string(fields.Type),
+				"company_size":       fields.EmployeeCount,
+				"company_name":       fields.CompanyName,
+				"job_title":          fields.JobTitle,
+				"have_sales_contact": fields.HaveSalesContact,
+			},
+		},
+	}
+	select {
+	case q.events <- []HubSpotEvent{createUser, sendUserEvent}:
+	default:
+		q.log.Error("Create user in HubSpot failed")
+	}
+}
+
+// EnqueueEvent for sending user behavioral event to HubSpot.
+func (q *HubSpotEvents) EnqueueEvent(email, eventName string, properties map[string]interface{}) {
+	eventName = strings.ReplaceAll(eventName, " ", "_")
+	eventName = strings.ToLower(eventName)
+	eventName = eventPrefix + "_" + eventName
+
+	newEvent := HubSpotEvent{
+		Endpoint: "https://api.hubapi.com/events/v3/send?hapikey=" + q.escapedAPIKey,
+		Data: map[string]interface{}{
+			"email":      email,
+			"eventName":  eventName,
+			"properties": properties,
+		},
+	}
+	select {
+	case q.events <- []HubSpotEvent{newEvent}:
+	default:
+		q.log.Error("Sending user event in HubSpot failed")
+	}
+}
+
+// handleSingleEvent for handle the single HubSpot API request.
+func (q *HubSpotEvents) handleSingleEvent(ctx context.Context, ev HubSpotEvent) (err error) {
+	payloadBytes, err := json.Marshal(ev.Data)
+	if err != nil {
+		return Error.New("json marshal failed: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, ev.Endpoint, bytes.NewReader(payloadBytes))
+	if err != nil {
+		return Error.New("new request failed: %w", err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	resp, err := q.httpClient.Do(req)
+	if err != nil {
+		return Error.New("send request failed: %w", err)
+	}
+
+	defer func() {
+		err = resp.Body.Close()
+	}()
+
+	return nil
+}
+
+// Handle for handle the HubSpot API requests.
+func (q *HubSpotEvents) Handle(ctx context.Context, events []HubSpotEvent) (err error) {
+	defer mon.Task()(&ctx)(&err)
+	for _, ev := range events {
+		err := q.handleSingleEvent(ctx, ev)
+		if err != nil {
+			return Error.New("error handling hubspot event: %w", err)
```

Comment:
Here `Error.New("handle event: %w", err)` would probably suffice, since the upper layers already add the `hubspot` information.

Similarly as before, since "Error" will be already included in the log line, then it's not necessary to include it.